src/lib/core/fiber.c

```diff
@@ -620,8 +628,25 @@ fiber_join(struct fiber *fiber)
 			 * to put it back in.
 			 */
 			rlist_add_tail_entry(&fiber->wake, fiber(), state);
-			fiber_yield();
-		} while (! fiber_is_dead(fiber));
+			if (timeout != TIMEOUT_INFINITY) {
+				double time = fiber_time();
+				exceeded = fiber_yield_timeout(timeout);
+				timeout -= (fiber_time() - time);
```

Comment:
Mb it is worth using ev_timer facilities?

Reply:
If you look at fiber_yield_timeout, it already uses ev_timers. So I reused it

Reply:
Ok, as you want, I don't insist on this change. My intention was to avoid conversion from ev_time to double (i.e. use more native way to do it).
Also let's rename exceeded to is_timedout :)

Reply:
It's quite strange to use fiber_time for timeouts. See https://github.com/tarantool/tarantool/commit/d988d7fb92fe1dda4b64218fb06813e93eb56ed1:
```
fiber_time() reports real time, which shouldn't be used for calculating
timeouts as it is affected by system time changes.
```

Reply:
@olegrok good remark. @mechanik20051988 please elaborate on that.